Prototope/Gesture.swift

```diff
@@ -593,19 +633,63 @@ public struct TouchSequence<I: Printable> : SampleSequenceType {
     }
 }
 
+// MARK: Gesture-to-gesture interaction
+
+//Need to have a way to map Gestures to UIGestureRecognizers
+var gestureMap = [UIGestureRecognizer:GestureType]()
+
+func gestureForGestureRecognizer(gestureRecognizer: UIGestureRecognizer) -> GestureType? {
+    return gestureMap[gestureRecognizer]
+}
+
+@objc class GestureRecognizerBridge: NSObject, UIGestureRecognizerDelegate {
+    let gesture: GestureType
+    
+    init(_ gesture: GestureType) {
+        self.gesture = gesture
+        super.init()
+        gesture.underlyingGestureRecognizer.delegate = self
+    }
+    
+    func gestureRecognizer(gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWithGestureRecognizer otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        if let otherGesture = gestureForGestureRecognizer(otherGestureRecognizer) {
+            return gesture.shouldRecognizeSimultaneouslyWithGesture(otherGesture)
+        }
+        return false
+    }
+}
+
 // MARK: - Internal interfaces
 
-public protocol GestureType: _GestureType {}
+public protocol GestureType: _GestureType {
+    var shouldRecognizeSimultaneouslyWithGesture: GestureType -> Bool { get set }
+}
 
-private func handleTransferOfGesture(gesture: UIGestureRecognizer, fromLayer: Layer?, toLayer: Layer?) {
+private func handleTransferOfGesture(gesture:GestureType, fromLayer: Layer?, toLayer: Layer?) {
+    let recognizer = gesture.underlyingGestureRecognizer
+    
+    switch (fromLayer, toLayer) {
+    case (.None, .Some):
+        gestureMap[recognizer] = gesture
+    case (.Some, .None):
+        gestureMap.removeValueForKey(recognizer)
+    default:
+        ()
+    }
+    
 	if fromLayer !== toLayer {
-		fromLayer?.view.removeGestureRecognizer(gesture)
-		toLayer?.view.addGestureRecognizer(gesture)
+		fromLayer?.view.removeGestureRecognizer(recognizer)
+		toLayer?.view.addGestureRecognizer(recognizer)
 	}
 }
 
 public protocol _GestureType {
 	weak var hostLayer: Layer? { get nonmutating set }
+    var underlyingGestureRecognizer: UIGestureRecognizer { get }
```

Comment:
I think I was trying to avoid this too, out of a sense of idealism (that maybe you'd want to define a gesture that is not a `UIGestureRecognizer`), but it makes way more sense to be pragmatic here.